internal/service/dtc/dtc_record_a_resource_test.go

```diff
@@ -0,0 +1,389 @@
+					// Test fields with default value
+					resource.TestCheckResourceAttr(resourceName, "disable", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_ttl", "false"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccDtcRecordAResource_disappears(t *testing.T) {
+	resourceName := "nios_dtc_record_a.test"
+	var v dtc.DtcRecordA
+	ipv4addr := acctest.RandomIP()
+	serverName := acctest.RandomNameWithPrefix("dtc-server")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckDtcRecordADestroy(context.Background(), &v),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDtcRecordABasicConfig(serverName, ipv4addr),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDtcRecordAExists(context.Background(), resourceName, &v),
+					testAccCheckDtcRecordADisappears(context.Background(), &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccDtcRecordAResource_Comment(t *testing.T) {
+	var resourceName = "nios_dtc_record_a.test_comment"
+	var v dtc.DtcRecordA
+	ipv4addr := acctest.RandomIP()
+	serverName := acctest.RandomNameWithPrefix("dtc-server")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDtcRecordAComment(serverName, ipv4addr, "This is a comment"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDtcRecordAExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "comment", "This is a comment"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccDtcRecordAComment(serverName, ipv4addr, "This is an updated comment"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDtcRecordAExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "comment", "This is an updated comment"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccDtcRecordAResource_Disable(t *testing.T) {
+	var resourceName = "nios_dtc_record_a.test_disable"
+	var v dtc.DtcRecordA
+	ipv4addr := acctest.RandomIP()
+	serverName := acctest.RandomNameWithPrefix("dtc-server")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDtcRecordADisable(serverName, ipv4addr, false),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDtcRecordAExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "disable", "DISABLE_REPLACE_ME"),
```

Comment:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "disable", "false"),
```
Please make this change for update test case as well